Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/server/DccTicketingServer.kt

```diff
@@ -85,4 +85,19 @@ class DccTicketingServer @Inject constructor(
         val certificateChain = handshake?.peerCertificates ?: emptyList()
         serverCertificateChecker.checkCertificate(certificateChain, jwkSet)
     }
+
+    suspend fun getAccessToken(
```

Comment:
You should validate the raw response with cert pinning in here, not sure if this will work with an Retrofit Response because it strips of the server certs.

Reply:
I guess cert pins shouldn't be checked in the retrofit response but rather using [CertificatePinner](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-certificate-pinner/) in OkHttp client.

Reply:
https://tenor.com/Vm1U.gif

Reply:
Acutally, it should. The problem with their cert pinner check is, that it differs from the one described in our TechSpec. In their check it is also not possible to change the trusted certs dynamically, which we get from the jwks. Furthermore, their check happens on every call, while the check **should only happen on certain calls**.

Reply:
yes, it only happens in the call where you include httpClient with the CertificatePinner and you can set up cert pinning before creating okhttpclient. but maybe let's have a call or discuss it in the chat

Reply:
it could be resolved later after https://github.com/corona-warn-app/cwa-app-android/pull/4415 is finalized and merged